sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/AggregationQuerySuite.scala

```diff
@@ -1018,6 +1018,16 @@ abstract class AggregationQuerySuite extends QueryTest with SQLTestUtils with Te
     val agg2 = agg1.groupBy($"text").agg(sum($"avg_res"))
     checkAnswer(agg2, Row("a", BigDecimal("11.9999999994857142860000")))
   }
+
+  test("SPARK-29122: hash-based aggregates for unfixed-length decimals in the interpreter mode") {
+    withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false",
+        SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.NO_CODEGEN.toString) {
+      withTable("t") {
+        spark.range(3).selectExpr("CAST(id AS decimal(38, 0)) a").write.saveAsTable("t")
```

Comment:
nit: it's better to use temp view if it can reproduce the bug, which can make the test faster.

Reply:
ah, ok. I'll update now.